test/tests/functional/pbs_snapshot_unittest.py

```diff
@@ -186,8 +189,10 @@ def take_snapshot(self, acct_logs=None, daemon_logs=None,
         if hosts is not None:
             hosts_str = ",".join(hosts)
             snap_cmd.append("--additional-hosts=" + hosts_str)
+        if primary_host is not None:
+            snap_cmd.append("-H " + primary_host)
 
-        ret = self.du.run_cmd(cmd=snap_cmd, logerr=False)
+        ret = self.du.run_cmd(cmd=snap_cmd, logerr=False, as_script=True)
```

Comment:
Why do you need as_script?

Reply:
It was failing without it, I'm not entirely sure why. Maybe because pbs_snapshot runs some commands with sudo? Again, I'm not really sure.